internal/compiler/wasm/optimizations.go

```diff
@@ -242,3 +247,44 @@ func reach(cg map[uint32][]uint32, keep map[uint32]struct{}, node uint32) {
 		}
 	}
 }
+
+// skipElem determines if a function in the table is really required:
+// We'll exclude anything with a prefix of "re2::" if none of the known
+// entrypoints into re2 are used.
+func (c *Compiler) skipElem(keep map[uint32]struct{}, idx uint32) bool {
+	if c.usesRE2(keep) {
+		c.debug.Printf("using re2, not skipping its elements")
+		return false
+	}
+	return c.nameContains(idx, "re2::", "lexer::", "std::", "__cxa_pure_virtual", "operator", "parser_")
```

Comment:
This is a little brittle... but it is the correct list of substrings _now_, we'll have to be careful when changing things in the future.

Reply:
I suppose the alternative would be to enumerate the functions in re2 and then check against that index here...not sure how much trouble that would be. It would certainly avoid potential issues w/ other C++ code in the future.

 I'm fine with leaving this as is for now. We just have to be careful when bringing other C++ code into the runtime. We'd discover the issue w/ tests as long as those tests didn't include policies that rely on re2...